dali/operators/imgcodec/image_decoder.h

```diff
@@ -639,16 +639,14 @@ class ImageDecoder : public StatelessOperator<Backend> {
 
     if (st.need_processing) {
       if constexpr (std::is_same<MixedBackend, Backend>::value) {
-        // This is a workaround for nvImageCodec <= 0.2
-        auto alloc_stream =
-            need_host_sync_alloc() ? mm::host_sync : st.image_info.cuda_stream;
-        st.device_buf = mm::alloc_raw_async_unique<uint8_t, mm::memory_kind::device>(
-            image_buffer_size, alloc_stream, st.image_info.cuda_stream);
-        st.image_info.buffer = st.device_buf.get();
+        // with nvImageCodec <= 0.2 we need to alloc on host order
+        st.device_buf.Resize(TensorShape<1>{image_buffer_size}, DALI_UINT8);
```

Comment:
Host-ordered resize is _generally_  unsafe - it's safe now because we wait (on host) for the previous invocation of the operation to end. This is, however, rather incidental and I don't think we should endorse or even accept that.